### DISCUSSION TITLE
v0.3.3 planning and discussion

### DISCUSSION THREAD

**Discussion by kheina** [kheina-com/Blue-Blocker#148] [General]
well, now that the repo is finally updated to typescript, we can start doing other stuff! again, huge thanks to @jaredkotoff, it wouldn't have even been an idea without you doing most of the work on your own.

some basic ideas that I'm going to look into:
1. promoted tweets and how they'll be handled and interact with user settings
2. some code cleanup things about where the travis db code lives (it occurred to me later that it should be in the background folder and not utils, but I didn't want to block 0.3.0 going out)
3. context menu cleanup from #141, I just think it's getting messy
4. I think import/export of queue and safelist/unblocked would be really cool to add for people switching over from [bluesblocker](https://chrome.google.com/webstore/detail/bluesblocker/dbcfmkldigmecjpjeojbcfbjidmcngfe) since the creator is [retiring the extension](https://github.com/kheina-com/Blue-Blocker/pull/60#issuecomment-1531054336)

and then anything else you guys want or suggest, I know @MrAwesome was working on #120 and was waiting for TS stuff to be done, and also wanted #133

this isn't a job though, do whatever the hell you want

**Comment by MrAwesome** [comment]
Awesome! Very exciting. I'll get started on redoing #120 this week hopefully, just got a few Life Things to deal with and then I should have plenty of time.

**Comment by jaredcat** [comment]
I was wanted to add testing and think saw @MrAwesome say this too. Do have opinions on which framework we want to use?
Feel like jest is probably the most straight forward.

Also @DanielleMiu maybe setup GitHub Project for this so we can have a nice kanban view of tasks

**Reply by kheina** [reply]
jest is the only one I've heard of but tbh I haven't used any js testing frameworks

**Reply by MrAwesome** [reply]
Jest sounds good to me. I've used it a fair amount in personal projects, it's pretty straightforward (as much as any js testing framework can be, anyway)

**Comment by kheina** [comment]
ok so, so far I've published 2 new versions, so this discussion is a little out of date, but I've mostly been running around fixing things from 0.3.0. really kicking myself about that manifest change too.

> Also @DanielleMiu maybe setup GitHub Project for this so we can have a nice kanban view of tasks

I guess I'm not familiar with github projects, I'll look into that tomorrow evening after work. I didn't see it until now
